src/licensedcode/models.py

```diff
@@ -772,6 +775,160 @@ def get_rules(
     return chain(licenses_as_rules, rules)
 
 
+def get_license_dirs(
+    additional_dirs,
+):
+    """
+    Takes in a list of additional directories specified during license detection
+    and produces a list of all the subdirectories containing license files.
+    """
+    # convert to absolute path in case user passes in a relative path, which messes up building rules from licenses
+    return [f"{str(Path(path).absolute())}/licenses" for path in additional_dirs]
+
+
+def get_rule_dirs(
+    additional_dirs,
+):
+    """
+    Takes in a list of additional directories specified during license detection
+    and produces a list of all the subdirectories containing rule files.
+    """
+    return [f"{str(Path(path).absolute())}/rules" for path in additional_dirs]
+
+
+def get_paths_to_installed_licenses_and_rules():
+    """
+    Returns a list of paths to externally packaged licenses or rules that the user has
+    installed. Gets a list of all of these licenses (installed as plugins) and
+    then gets the plugins containing licenses by checking that their names start
+    with a common prefix.
+    """
+    from importlib_metadata import entry_points
+    from plugincode.location_provider import get_location
+    installed_plugins = entry_points(group='scancode_location_provider')
+    paths = []
+    for plugin in installed_plugins:
+        if plugin.name.startswith(EXTERNAL_LICENSE_PLUGIN_PREFIX):
+            # get path to directory of licenses and/or rules
+            location_key = plugin.name
+            paths.append(get_location(location_key))
+    return paths
+
+
+def load_licenses_from_multiple_dirs(
+    license_directories,
+    with_deprecated=False,
+):
+    """
+    Takes in a list of directories containing additional licenses to use in
```

Comment:
As a general style rule, we prefer:
1. imperative style
2. starting with the outcome

so here, this would become:
```
Return a mapping of {key: License} built from a list of ``license_directories``.
```
... which raises a few extra questions: ```license_directories``: are these absolute locations? say so.